### ISSUE TITLE
Improved layout of chat messages (square corners)

### ISSUE THREAD

**Issue by IanMayo** [serge-web/serge#1687]
When there is a series of chat messages we could use a square corner for the "adjacent" messages, like in this screenshot:
![image](https://user-images.githubusercontent.com/1108513/180484027-fd1a681d-8f0c-4296-ab3b-6573e703f05f.png)

I'll try to do the logic to recognise adjacent messages, including whether a message needs square markers at top and/or bottom.

I'm waiting to hear from @ShendMikullovci about if the right-hand side should also be a square corner. That seems the _Consistent_ way.

**Comment by ShendMikullovci** [comment]
Hi Ian! I was sure I replied to this issue! Apologies.

The reason why Messenger went with a left-only angle curvature is to align the messages to a.) themselves and b.) to the interlocutor. This way, you don't have to have an icon/logo/picture by each message, but rather display it at the top of the message list that they have sent, or at the bottom. 

That's why I suggest we stick to left only.

**Comment by IanMayo** [comment]
Hello @ShendMikullovci - you did reply, in a different issue.  Your logic is fine.

**Comment by IanMayo** [comment]
Aah, I've found an example of just rounding one half of the messages.  This is how Signal displays messages:
![image](https://user-images.githubusercontent.com/1108513/189883809-f5dfa9cc-af17-4be0-856b-f72aa5d5e1c5.png)

**Comment by ShendMikullovci** [comment]
Yes, exactly! Side curvature adds another grouping factor on top of color and side of display, further emphasizing to whom the messages belong.
